### PR TITLE
Add a recipe for CUDA RPC

### PR DESCRIPTION
**Pull request by mrshenli**

### CODE REVIEW

File: recipes_source/zero_redundancy_optimizer.rst [hunk 1]
```diff
@@ -1,8 +1,7 @@
 Shard Optimizer States with ZeroRedundancyOptimizer
 ===================================================
 
-.. note:
-    `ZeroRedundancyOptimizer` is introduced in PyTorch 1.8 as a prototype
+.. note:: `ZeroRedundancyOptimizer` is introduced in PyTorch 1.8 as a prototype
```
Comment (mrshenli) [inline recipes_source/zero_redundancy_optimizer.rst:4]: I saw that this note wasn't displayed correctly. This edit should fix it.

File: recipes_source/cuda_rpc.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+- PyTorch 1.8+
+- `Getting Started With Distributed RPC Framework <https://pytorch.org/tutorials/intermediate/rpc_tutorial.html>`_
+
+
+What is CUDA RPC?
+------------------------------------
+
+CUDA RPC supports directly sending Tensors from local CUDA memory to remote
+CUDA memory. Prior to v1.8 release, PyTorch RPC only takes CPU Tensors. As a
```
Comment (beauby) [inline recipes_source/cuda_rpc.rst:24]:
```suggestion
CUDA memory. Prior to v1.8 release, PyTorch RPC only accepts CPU Tensors. As a
```

File: recipes_source/cuda_rpc.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+---------------------------------------
+
+The code below shows how to use CUDA RPC. The model contains two linear layers
+and is split into two shards. The two shards are placed on ``worker0`` and
+``worker1`` respectively, and ``worker0`` serves as the master that drives the
+forward and backward passes. Note that we intentionally skipped
+`DistributedOptimizer <https://pytorch.org/docs/master/rpc.html#module-torch.distributed.optim>`_
+to highlight the performance improvements when using CUDA RPC. The experiment
+repeats the forward and backward passes 10 times and measures the total
+execution time. It Compares uring CUDA RPC against using CPU RPC.
```
Comment (beauby) [inline recipes_source/cuda_rpc.rst:58]:
```suggestion
execution time. It compares using CUDA RPC against using manually staging to CPU memory and using CPU RPC.
```

**Review by beauby — APPROVED** [review]

**Comment by mrshenli** [comment]
screencapture-deploy-preview-1429-pytorch-tutorials-preview-netlify-app-recipes-cuda-rpc-html-2021-03-19-17_44_13

**Comment by mrshenli** [comment]
Hey @brianjo, shall we land this recipe?

**Comment by brianjo** [comment]
I’ll do it right now. Thanks!


________________________________

**Comment by brianjo** [comment]
Published here: https://pytorch.org/tutorials/recipes/cuda_rpc.html

**Pull request merged**